crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -538,16 +543,13 @@ impl AlgorithmUpdaterV1 {
             )?;
             total = total.saturating_add(bytes as u128);
         }
+        self.unrecorded_blocks_bytes = self.unrecorded_blocks_bytes.saturating_sub(total);
```

Comment:
I'm thinking about the following scenario, not sure if this is an issue or not:
1. we have 3 blocks in `self.unrecorded_blocks`
2. we `remove()` 2 of them, but we get an error on 3rd one
3. now, the `unrecorded_blocks_bytes` becomes out-of-sync